### PR TITLE
Heal upgrade

### PR DESCRIPTION
**Pull request by HYBleek**
I have tested my code in branch testing-with-all-feature, and I work well.
There are only four things to be changed. Except for the EssenceUse package, I also make changes to the input boundary, the driver and the controller

### CODE REVIEW

File: src/main/java/usecase_essence_use/data_preset_normal/collectibleNeedSetting.java [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
+package usecase_essence_use.data_preset_normal;
 
-public class CalculatorCollectible {
-    public int essenceHPExchangeRate;
-    public int essenceForUpgrade;
-    public CalculatorCollectible(){
+public final class collectibleNeedSetting {
+    public final int essenceHPExchangeRate;
+    public final int essenceForUpgrade;
+    public collectibleNeedSetting(){
```
Comment (littlepeach17) [inline src/main/java/usecase_essence_use/data_preset_normal/collectibleNeedSetting.java:6]: javadoc needed for constructor

File: src/main/java/usecase_essence_use/essenceUseKey/Upgrader.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    private final essenceUseSpeakerManager speaker;
+
+    /**
+     * The constructor of healing. This class is to be triggered by the HealingUpgradingControl. And It will recover the
+     * certain amount of HP that is determined by the HealInfo
+     * @param player the player
+     * @param Manager the healCalculator that the trigger connect to
+     * @param trigger the trigger word, usually a big capital letter
+     */
+
```
Comment (littlepeach17) [inline src/main/java/usecase_essence_use/essenceUseKey/Upgrader.java:27]: javadoc mistake, constructor is for upgrading not healing

**Comment by HongyiZhan** [comment]
approve

**Review by HongyiZhan — APPROVED** [review]
good

**Pull request merged**